src/pie-chart/utils.ts

```diff
@@ -163,23 +164,70 @@ export const balanceLabelNodes = (
     node.setAttribute('transform', '');
 
     // Calculate how much the current node is overlapping with the previous one.
-    const offset = previousBBox.y + previousBBox.height + MARGIN - box.y;
+    const yOffset = previousBBox.y + previousBBox.height + MARGIN - box.y;
 
-    if (offset > 0) {
+    if (yOffset > 0) {
+      const xOffset = computeXOffset(box, yOffset, radius) * (leftSide ? -1 : 1);
       // Move the label down.
-      node.setAttribute('transform', `translate(0 ${offset})`);
+      node.setAttribute('transform', `translate(${xOffset} ${yOffset})`);
 
       // Adjust the attached line accordingly.
       const lineNode = node.parentNode?.querySelector(`.${styles['label-line']}`);
       if (lineNode) {
-        const { endY } = marker;
-        lineNode.setAttribute('y2', '' + (endY + offset));
+        const { endY, endX } = marker;
+        lineNode.setAttribute('y2', '' + (endY + yOffset));
+        lineNode.setAttribute('x2', '' + (endX + xOffset));
       }
 
       // Update the position accordingly to inform the next label
-      box.y += offset;
+      box.y += yOffset;
+      box.x += xOffset;
     }
 
     previousBBox = box;
   }
 };
+
+const squareDistance = (edge: Array<number>): number => Math.pow(edge[0], 2) + Math.pow(edge[1], 2);
+
+const computeXOffset = (box: { x: number; y: number; height: number }, yOffset: number, radius: number): number => {
```

Comment:
Could we add some unit tests to this computeXOffset?

Reply:
There are some snapshot tests for it